pkg/text/text.go

```diff
@@ -0,0 +1,178 @@
+			segment = nil
+		}
+		segment = append(segment, unicode.ToLower(r))
+	}
+	output = addSegment(output, segment)
+	return string(output)
+}
+
+func addSegment(inrune, segment []rune) []rune {
+	if len(segment) == 0 {
+		return inrune
+	}
+	if len(inrune) != 0 {
+		inrune = append(inrune, '-')
+	}
+	inrune = append(inrune, segment...)
+	return inrune
+}
+
+// Title returns a copy of the string s with all Unicode letters that begin words mapped to their Unicode title case.
+func Title(s string) string {
+	c := cases.Title(language.English)
+	return c.String(s)
+}
+
+// RemoveExcessiveWhitespace returns a copy of the string s with excessive whitespace removed.
+func RemoveExcessiveWhitespace(s string) string {
+	return whitespaceRE.ReplaceAllString(strings.TrimSpace(s), " ")
+}
+
+// DisplayWidth calculates what the rendered width of string s will be.
+func DisplayWidth(s string) int {
+	return ansi.PrintableRuneWidth(s)
+}
+
+// Truncate returns a copy of the string s that has been shortened to fit the maximum display width.
+func Truncate(maxWidth int, s string) string {
+	w := DisplayWidth(s)
+	if w <= maxWidth {
+		return s
+	}
+	tail := ""
+	if maxWidth >= minWidthForEllipsis {
+		tail = ellipsis
+	}
+	r := truncate.StringWithTail(s, uint(maxWidth), tail)
+	if DisplayWidth(r) < maxWidth {
+		r += " "
+	}
+	return r
+}
+
+// TruncateMultiline returns a copy of the string s that has been shortened to fit the maximum
+// display width. If string s has multiple lines the first line will be shortened and all others
+// removed.
+func TruncateMultiline(maxWidth int, s string) string {
+	if i := strings.IndexAny(s, "\r\n"); i >= 0 {
+		s = s[:i] + ellipsis
+	}
+	return Truncate(maxWidth, s)
+}
+
+// Pluralize returns a concatenated string with num and the plural form of thing if necessary.
+func Pluralize(num int, thing string) string {
+	if num == 1 {
+		return fmt.Sprintf("%d %s", num, thing)
+	}
+	return fmt.Sprintf("%d %ss", num, thing)
+}
+
+func fmtDuration(amount int, unit string) string {
+	return fmt.Sprintf("about %s ago", Pluralize(amount, unit))
+}
+
+// FuzzyAgo returns a human readable string of the time duration between a and b that is estimated
+// to the nearest unit of time.
+func FuzzyAgo(a, b time.Time) string {
+	ago := a.Sub(b)
+
+	if ago < time.Minute {
+		return "less than a minute ago"
+	}
+	if ago < time.Hour {
+		return fmtDuration(int(ago.Minutes()), "minute")
+	}
+	if ago < 24*time.Hour {
+		return fmtDuration(int(ago.Hours()), "hour")
+	}
+	if ago < 30*24*time.Hour {
+		return fmtDuration(int(ago.Hours())/24, "day")
+	}
+	if ago < 365*24*time.Hour {
+		return fmtDuration(int(ago.Hours())/24/30, "month")
+	}
+
+	return fmtDuration(int(ago.Hours()/24/365), "year")
+}
+
+// FuzzyAgoAbbr is an abbreviated version of FuzzyAgo. It returns a human readable string of the
+// time duration between a and b that is estimated to the nearest unit of time.
+func FuzzyAgoAbbr(a, b time.Time) string {
```

Comment:
I would be open to new naming here.

Reply:
I feel like we should have only a single implementation of FuzzyAgo, to reduce confusion. Since the output of this function is very terse, we could drop it in favor of the first implementation